src/Avalonia.Controls/DateTimePickers/TimePicker.cs

```diff
@@ -230,11 +308,12 @@ private void SetSelectedTimeText()
                 {
                     var hr = newTime.Hours;
                     hr = hr > 12 ? hr - 12 : hr == 0 ? 12 : hr;
-                    newTime = new TimeSpan(hr, newTime.Minutes, 0);
+                    newTime = new TimeSpan(hr, newTime.Minutes, newTime.Seconds);
```

Comment:
The current behaviour with a time that includes seconds when `UseSeconds = false` is to keep the seconds, which may be misleading, particular as the time span defaults to the current time, so a user can select a time, but be off by 0-59 seconds without knowing it.

I think it would be better to exclude the seconds when `UseSeconds = false`; I'm not sure what would be the right behaviour when setting `UseSeconds = true` having already been `false`, but my instinct is that it should be cleared to zero.

Reply:
Good catch.

Interestingly (in the absence of it being set manually with `UseSeconds == true`), `newTime.Seconds` appears to be set to be equal to the current time when the Control is first clicked on (not when it is first loaded or when the check button is clicked to set the time) and then stays that way forever (doesn't update even when the Control is clicked again or another control is clicked and then this control is clicked again).  I'm not, currently, entirely sure why this is and it also isn't super relevant to this particular issue.

Changing the modified line to
```
newTime = (UseSeconds) ? 
                        new TimeSpan(hr, newTime.Minutes, newTime.Seconds) :
                        new TimeSpan(hr, newTime.Minutes, 0);
```
causes 0 to be used for seconds when `UseSeconds == false` & the selected seconds to be used for seconds when `UseSeconds == true`.  But it doesn't, alone, address the case where `UseSeconds` switches from `false` to `true`.

Regarding when `UseSeconds == false` changes to `UseSeconds == true`.  That is also a good edge case you've identified.  We can detect this case in the `UseSeconds` portion of the `OnPropertyChanged` override method.
We can detect that `UseSecond` went from `false` to `true` and pass an argument to `SetSelectedTimeText()` to indicate that 0 should be used for seconds.

It could look something like this which should result in
- using a 0 for seconds if `UseSeconds == false`
- using a 0 for seconds if `UseSeconds == true` but we came this method because `UseSeconds` went from `false` to `true`
- otherwise (if `UseSeconds == true` but was not just changed from `false`) using the SelectedTime.Seconds for seconds
![image](https://github.com/user-attachments/assets/b8be07a7-6b39-4909-bda3-8aff4061f2da)

Reply:
Having messed around a bit, I'm still not sure what the correct behaviour should be. I'm happy enough with the behaviour of us setting the `seconds` to zero when completing an interaction; trivial change is in https://github.com/VisualMelon/Avalonia/commit/7e4df0f557eebac661246db7f6d4914ab32d843d along with trivial example. It doesn't exercise the scenario where you have another control/piece of code updating the time while `UseSeconds` is false, and then setting it to true, but again, not sure what the behaviour should be, so may well warrant something more interesting

Reply:
Sorry I've not been more active on this; we should probably create an issue for it, as it is a serious breaking change and don't want it to get lost: I'll try to remember to do so on the weekend if no-one beats me to it.